### ISSUE TITLE
byte length error when Deno upgraded to v1.31.2

### ISSUE THREAD

**Issue by halparn** [lideming/btrdb#2]
Hi, for a while i'm using btrdb databeses for simple data storage in Deno. After i upgrade Deno it's last version, and try to creating table and upsert some initial datas in it I encountered this error message: 

`
   error: Uncaught Error: Expect 16 == 19
      throw new Error("Expect " + r.written + " == " + len);
            ^
    at writeValue (https://deno.land/x/btrdb@v0.8.3/src/utils/binval.ts:75:13)
    at writeValue (https://deno.land/x/btrdb@v0.8.3/src/utils/binval.ts:107:9)
    at DocumentValue.writeTo (https://deno.land/x/btrdb@v0.8.3/src/utils/value.ts:141:5)
    at InFileStorage.addData (https://deno.land/x/btrdb@v0.8.3/src/pages/storage.ts:325:11)
    at async DbDocSet._set (https://deno.land/x/btrdb@v0.8.3/src/database/DbDocSet.ts:150:13)
    at async DbDocSet.upsert (https://deno.land/x/btrdb@v0.8.3/src/database/DbDocSet.ts:114:5)
`

Here is an example code how i offently upsert data usage:

```
const exampleWriteCode = async (dbpath, data) => {
  const db = new Database();
  await db.openFile(dbpath);
  const tbl = await db.createSet("tbl", "doc");
  await db.commit();

  for (let i = 0; i < data.length; i++) {
    const d = data[i];
    await tbl.upsert({ ...d });
  }

  await db.forceCommit();
};

```

After i encountered this error, downgraded deno to 1.31.1 fix this issue. This error produced v1.31.2 or later.

**Comment by lideming** [comment]
Hi @halparn, thank you for reporting.
~~Unfortunately I can't reproduce this error. Could you provide a document object that could be the cause of this error?~~

**Comment by lideming** [comment]
It's a bug of `TextEncoder.encodeInto()` in Deno Runtime: https://github.com/denoland/deno/issues/18255

**Comment by lideming** [comment]
Fixed in https://github.com/denoland/deno/pull/18518 and released in [Deno v1.32.2](https://github.com/denoland/deno/releases/tag/v1.32.2)
